### PR TITLE
Feature/docs base

### PR DESCRIPTION
**Pull request by Sypheos**
#### Summary:
This pull request introduce a new documentation organisation and workflow.
We will now use Hugo to build the documentation website. It allow us to have the documentation inside the same repository and build a rich documentation but it is also easy to run locally.
The online documentation will automatically be published on the project github page (via the gh-pages branch) by travis on each commit and pull request made on the master branch.
The gh-pages will have a static site by tagged version. An index.html at the root will redirect to the lasted documentation.

Reference #401
Reference #206 

#### Changes
Change the doc file tree into a hugoable tree.
Add hugo-theme-doc as a theme submodule until we have out own theme.
Add docs-gen, docs-server, docs-deps rules to the Makefile.
Add new job in .travis.yml to build the doc

#### Notes for Reviewers

NO documentation content will be added or discussed in this PR.

#### Release Notes

* Add a new documentation system based on [Hugo|https://gohugo.io/]

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -126,3 +130,12 @@ after_success:
       GO111MODULE=on go run github.com/goreleaser/goreleaser --snapshot --release-notes <(go run github.com/TheThingsIndustries/release-notes -owner TheThingsNetwork -repo lorawan-stack -id "release-notes" -head $(git rev-parse HEAD))
     fi
   fi
+- if [[ "$RUNTYPE" == "release" ]]; then $MAGE docs:gen; fi
+deploy:
+- provider: pages
+  local_dir: doc/public
+  skip_cleanup: true
+  github_token: $GITHUB_TOKEN
```
Comment (htdvisser) [inline .travis.yml]: Note to self: Set this token in Travis is not already set

File: doc/config.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  description = "The Things Network Stack for LoRaWAN is an open-source LoRaWAN network stack suitable for large, global and geo-distributed public and private networks as well as smaller networks."
+  github_repository = "https://github.com/TheThingsNetwork/lorawan-stack"
+  version= "v3.0.3"
+  # Documentation repository section
+  # documentation repository (set edit link to documentation repository)
+  github_doc_repository = "https://github.com/TheThingsNetwork/lorawan-stack"
+  source_url = "https://github.com/TheThingsNetwork/lorawan-stack/blob/master/"
+
+[[menu.main]]
+  name = "Main"
```
Comment (johanstokking) [inline doc/config.toml]: What is this title for, all of the documentation? Will there be other things here, and if so, what?

Screenshot 2019-07-10 at 17 01 31

Like, can we make `Main` into a Stack landing page and have the actual documentation in its own section?

File: doc/config.toml.tmpl [hunk 3]
```diff
@@ -0,0 +1,32 @@
+baseURL = "http://localhost/"
+languageCode = "en-us"
+title = "The Things Network Stack for LoRaWAN"
```
Comment (johanstokking) [inline doc/config.toml.tmpl]: Let's explicitly call this documentation for SEO, unless we make this a general Stack landing page

File: doc/content/_index.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+title: "Home"
```
Comment (johanstokking) [inline doc/content/_index.md]: And then this is `Overview` of what to be found here

File: doc/content/_index.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+description: ""
+weight: 1
+---
+
+> TODO: Import content from README.md
```
Comment (rvolosatovs) [inline doc/content/_index.md]: issue reference missing

File: doc/content/gettingstarted/_index.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+title: "Quick Start"
+description: ""
+weight: 2
+---
+
+> TODO: Move content of `gettingstarted.md` to this section
```
Comment (rvolosatovs) [inline doc/content/gettingstarted/_index.md]: IDEM

File: doc/config.toml (hunk 2 above)
Comment (htdvisser) [reply doc/config.toml]: Removed, it's now just a page link

File: doc/content/_index.md (hunk 4 above)
Comment (htdvisser) [reply doc/content/_index.md]: Renamed

**Comment by htdvisser** [comment]
image

This is it. The "Home" on the left comes from the theme, otherwise it would also be "Overview".

**Pull request merged**